src/PlaywrightEnvironment.ts

```diff
@@ -64,154 +59,162 @@ const getBrowserPerProcess = async (
+          // Run a debugger (in case Playwright has been launched with `{ devtools: true }`)
+          await this.global.page.evaluate(() => {
+            // eslint-disable-next-line no-debugger
+            debugger
+          })
+          // eslint-disable-next-line no-console
+          console.log('\n\n🕵️‍  Code is paused, press enter to resume')
+          // Run an infinite promise
+          return new Promise((resolve) => {
+            const { stdin } = process
+            const listening = stdin.listenerCount('data') > 0
+            const onKeyPress = (key: string): void => {
+              if (
+                key === KEYS.CONTROL_C ||
+                key === KEYS.CONTROL_D ||
+                key === KEYS.ENTER
+              ) {
+                stdin.removeListener('data', onKeyPress)
+                if (!listening) {
+                  if (stdin.isTTY) {
+                    stdin.setRawMode(false)
+                  }
+                  stdin.pause()
                 }
-                stdin.pause()
+                resolve()
               }
-              resolve()
             }
-          }
-          if (!listening) {
-            if (stdin.isTTY) {
-              stdin.setRawMode(true)
+            if (!listening) {
+              if (stdin.isTTY) {
+                stdin.setRawMode(true)
+              }
+              stdin.resume()
+              stdin.setEncoding('utf8')
             }
-            stdin.resume()
-            stdin.setEncoding('utf8')
-          }
-          stdin.on('data', onKeyPress)
-        })
-      },
+            stdin.on('data', onKeyPress)
+          })
+        },
+      }
     }
-  }
 
-  async handleTestEvent(event: Event, state: State): Promise<void> {
-    // Hack to set testTimeout for jestPlaywright debugging
-    if (
-      event.name === 'add_test' &&
-      event.fn &&
-      event.fn.toString().includes('jestPlaywright.debug()')
-    ) {
-      // Set timeout to 4 days
-      state.testTimeout = 4 * 24 * 60 * 60 * 1000
+    async handleTestEvent(event: Event, state: State): Promise<void> {
+      // Hack to set testTimeout for jestPlaywright debugging
+      if (
+        event.name === 'add_test' &&
+        event.fn &&
+        event.fn.toString().includes('jestPlaywright.debug()')
+      ) {
+        // Set timeout to 4 days
+        state.testTimeout = 4 * 24 * 60 * 60 * 1000
+      }
     }
-  }
 
-  async teardown(jestConfig: JestConfig.InitialOptions = {}): Promise<void> {
-    const { page, context, browser } = this.global
-    if (page) {
-      page.removeListener('pageerror', handleError)
-    }
-    if (context) {
-      await context.close()
-    }
-    if (page) {
-      await page.close()
-    }
+    async teardown(jestConfig: JestConfig.InitialOptions = {}): Promise<void> {
+      const { page, context, browser } = this.global
+      if (page) {
+        page.removeListener('pageerror', handleError)
+      }
+      if (context) {
```

Comment:
maybe we can simplify the whole function in something like that. Its enough to only call browser.close().
```js
const wait = []
if (browser) {
wait.push(browser.close())
}
if (!jestConfig.watch && !jestConfig.watchAll && teardownServer) {
wait.push(teardownServer())
}
await Promise.all(...wait)
```